File: builder-pattern/java/src/Phone.java
package src;

// 产品类
public class Phone {
  private String name = "PhoneName";
  private Integer[] screen = { 0, 0 };
  private Integer gpuType = 0;

  public void setName(String name) {
    this.name = name;
  }

  public String getName() {
    return this.name;
  }

  public void setScreen(Integer[] screen) {
    this.screen = screen;
  }

  public Integer[] getScreen() {
    return this.screen;
  }

  public void setGpuType(Integer type) {
    this.gpuType = type;
  }

  public Integer getGpuType() {
    return this.gpuType;
  }

  public String toString() {
    return "[name=" + name + ", screen=" + screen.toString() + ", gpuType=" + gpuType + "]";
  }

}

File: builder-pattern/go/src/Phone.go
package src

type Phone struct {
  name    string
  screen  []int
  gpuType int
}

func (p *Phone) SetName(name string) {
  p.name = name
}

func (p *Phone) GetName() string {
  return p.name
}

func (p *Phone) SetScreen(screen []int) {
  p.screen = screen
}

func (p *Phone) GetScreen() []int {
  return p.screen
}

func (p *Phone) SetGpuType(gpuType int) {
  p.gpuType = gpuType
}

func (p *Phone) GetGpuType() int {
  return p.gpuType
}
